cst438_register/src/main/java/com/cst438/service/GradebookServiceMQ.java

```diff
@@ -43,6 +47,18 @@ public void receive(CourseDTOG courseDTOG) {
 		
 		//TODO  complete this method in homework 4
 		
+		CourseDTOG.GradeDTO gradeDTO = new CourseDTOG.GradeDTO();
+		
+		Enrollment e = enrollmentRepository.findByEmailAndCourseId(gradeDTO.student_email, courseDTOG.course_id);
+		
+		if (e != null) {
+			e.setCourseGrade(gradeDTO.grade);
+			enrollmentRepository.save(e);
+		}
+		else {
+			throw  new ResponseStatusException( HttpStatus.BAD_REQUEST, "Could not find enrollment by email: " + gradeDTO.student_email );
```

Comment:
Since this is rabbitMQ, there's no HTTP involved, which means you can't throw HTTP errors. If `e` is `null`, you just won't do anything.